content/tutorials/flutter-cryptocurrency-app.textile

```diff
@@ -0,0 +1,720 @@
+
+h3(#notify-ui). Notifying the UI of New Data
+
+To make it easier to consume the data we get from these Streams in the UI, we will create a new class that extends ChangeNotifier. In Flutter, the simplest way to get notified when anything changes is by using ChangeNotifier.
+
+```[flutter]
+class CoinUpdates extends ChangeNotifier {
+  CoinUpdates({this.name});
+  final String name;
+
+  Coin _coin;
+
+  Coin get coin => _coin;
+  updateCoin(value) {
+    this._coin = value;
+    notifyListeners();
+  }
+}
+```
+
+Any UI widget that registers a listener for this object will get a notification whenever it has to rebuild with new data.
+
+The update will happen when calling the @updateCoin()@ method, which will assign the new coin data to @_coin@, then call @notifyListeners()@.
+
+We chose to transform Stream events into **ChangeNotifier** updates because they are much easier to use in the UI as they always have a valid value and don’t care if they are listened to multiple times.
+
+h3. Subscribe to Ably Channels
+
+Let’s break down the previous function and explain more how we are subscribing to the cryptocurrency channel in 4 steps:
+
+1. Get the list of realtime channels for your account, then get a specific one by its name
+
+```[flutter]
+ably.RealtimeChannel channel = _realtime.channels.get('[product:ably-coindesk/crypto-pricing]$coinCode:usd');
+```
+
+In a real App it’s probably a good idea to check if we got the channel that we requested.
+
+2. Subscribe to the chosen channel
+
+```[flutter]
+final Stream<ably.Message> messageStream = channel.subscribe();
+```
+
+3. The returned type from @subscribe()@ is a @Stream<Message>@, which feels a bit odd because we just subscribed to something. In reality the @subscribe@ tells Ably to start transmitting data.
+
+4. So we register a listener for this message Stream, and use `where` to filter null values.
+
+```[flutter]
+messageStream.where((event) => event.data != null).listen((message) {});
+```
+
+As we never stop listening to the channels in this app we can ignore the @StreamSubscription@ that is returned from @listen()@.
+
+Inside the listener, whenever a new message arrives, we call `updateCoin()` and pass it a new Coin mapped from the Message data.
+
+The return type of this function is a @List<CoinUpdates>@ which has the same length as @_coinTypes@ and with a @CoinUpdates@ object for every currency defined in @coinTypes@. 
+
+To be safe in case that this function is called more than once we wrap the @for@ loop in an @if@ that checks if we already have made our channel subscription.
+
+Now we have finished setting up all we need for the service, it’s time to see how we will use it to show the prices graphs. 
+
+To be able to access our services easily we use the service locator package @get_it@. You could also use @package/provider@ or any other solution that you are comfortable with.
+
+The following diagram visualizes how the data will flow from Ably to our App’s UI.
```

Comment:
Is this here at the right place?

Reply:
I added it after we finished explaining the service functions so it gives a visualization of how both parts will be connected, do you think it would make more sense to put it at the beginning?